src/main/resources/react4xp/_entries/Table.jsx

```diff
@@ -0,0 +1,213 @@
+    if (rows) {
+      return rows.map((row, i) => {
+        return (
+          <tr key={i}>
+            { this.createCell(row) }
+          </tr>
+        )
+      })
+    }
+  }
+
+  createRowsBody(rows) {
+    if (rows) {
+      return rows.map((row, i) => {
+        return (
+          <tr key={i}>
+            { this.createBodyTh(row) }
+            { this.createBodyTd(row) }
+          </tr>
+        )
+      })
+    }
+  }
+
+  createBodyTh(row) {
+    return Object.keys(row).map(function(keyName, keyIndex) {
+      const value = row[keyName]
+      if (keyName === 'th') {
+        if (typeof value === 'string' | typeof value === 'number') {
+          return (
+            <th key={keyIndex}>{value}</th>
+          )
+        } else {
+          return (
+            <th key={keyIndex} className={value.class} rowSpan={value.rowspan} colSpan={value.colspan}>{value.content}</th>
+          )
+        }
+      }
+    })
+  }
+
+  createBodyTd(row) {
+    return Object.keys(row).map(function(keyName, keyIndex) {
+      const value = row[keyName]
+      if (keyName === 'td') {
+        if (typeof value === 'string' | typeof value === 'number') {
+          return (
+            <th key={keyIndex}>{value}</th>
+          )
+        } else {
+          if (Array.isArray(value)) {
+            return value.map((cellValue, i) => {
+              return (
+                <td key={i}>{cellValue}</td>
+              )
+            })
+          } else {
+            return (
+              <td key={keyIndex} className={value.class} rowSpan={value.rowspan} colSpan={value.colspan}>{value.content}</td>
+            )
+          }
+        }
+      }
+    })
+  }
+
+  createCell(row) {
+    return Object.keys(row).map(function(keyName, keyIndex) {
+      const value = row[keyName]
+      if (keyName === 'td') {
+        if (typeof value === 'string' | typeof value === 'number') {
+          return (
+            <td key={keyIndex}>{value}</td>
+          )
+        } else {
+          if (Array.isArray(value)) {
+            return value.map((cellValue, i) => {
+              return (
+                <td key={i}>{cellValue}</td>
+              )
+            })
+          } else {
+            return (
+              <td key={keyIndex} rowSpan={value.rowspan} colSpan={value.colspan}>{value.content}</td>
+            )
+          }
+        }
+      }
+      if (keyName === 'th') {
+        if (typeof value === 'string' | typeof value === 'number') {
+          return (
+            <th key={keyIndex}>{value}</th>
+          )
+        } else {
+          if (Array.isArray(value)) {
+            return value.map((cellValue, i) => {
+              return (
+                <th key={i}>{cellValue}</th>
+              )
+            })
+          } else {
+            return (
+              <th key={keyIndex} rowSpan={value.rowspan} colSpan={value.colspan}>{value.content}</th>
+            )
+          }
+        }
+      }
+    })
+  }
+
```

Comment:
Kanskje det kunne vært brukt noe lignende "createElement" til å lage en td eller th for å slippe dobbel kode?

Reply:
Ja det er endel av koden som jeg skal rydde i nå :)